zb-classifier.js

```diff
@@ -950,6 +959,401 @@ class ZigbeeClassifier {
+    };
+
+    maxCoolTargetProperty.updated = function() {
+      coolTargetProperty.updateMaximum();
+    };
+
+    coolTargetProperty.updated = function() {
+      maxCoolTargetProperty.setMinimum(this.value);
+      minCoolTargetProperty.setMaximum(this.value);
+    };
+
+    coolTargetProperty.updateMaximum = function() {
+      if (!maxCoolTargetProperty.hasOwnProperty('value') ||
+          !deadbandProperty.hasOwnProperty('value') ||
+          !heatTargetProperty.hasOwnProperty('value')) {
+        // We don't have enough information yet
+        return;
+      }
+      const max1 = maxCoolTargetProperty.value;
+      const max2 = heatTargetProperty.value - deadbandProperty.value;
+      coolTargetProperty.setMaximum(Math.max(max1, max2));
+    };
+
+    minCoolTargetProperty.updated = function() {
+      coolTargetProperty.setMinimum(this.value);
+    };
+
+    absMinCoolTargetProperty.updated = function() {
+      minCoolTargetProperty.setMinimum(this.value);
+    };
+
+    deadbandProperty.updated = function() {
+      heatTargetProperty.updateMinimum();
+      coolTargetProperty.updateMaximum();
+    };
+
+    this.addProperty(
+      node,                             // device
+      'hold',                           // name
+      {                                 // property description
+        '@type': 'BooleanProperty',
+        label: 'Hold',
+        type: 'boolean',
+      },
+      PROFILE_ID.ZHA,                   // profileId
+      hvacThermostatEndpoint,           // endpoint
+      CLUSTER_ID.HVACTHERMOSTAT,        // clusterId
+      'tempSetpointHold',               // attr
+      'setOnOffWriteValue',             // setAttrFromValue
+      'parseOnOffAttr',                 // parseValueFromAttr
+      CONFIG_REPORT_MODE
+    );
+
+    const uiCfgEndpoint =
+      node.findZhaEndpointWithInputClusterIdHex(
+        CLUSTER_ID.HVACUSERINTERFACECFG_HEX);
+
+    if (uiCfgEndpoint) {
+      this.addProperty(
+        node,                             // device
+        'units',                          // name
+        {                                 // property description
+          label: 'Units',
+          type: 'string',
+          enum: ['C', 'F'],
+        },
+        PROFILE_ID.ZHA,                   // profileId
+        uiCfgEndpoint,                    // endpoint
+        CLUSTER_ID.HVACUSERINTERFACECFG,  // clusterId
+        'tempDisplayMode',                // attr
+        'setWriteEnumValue',              // setAttrFromValue
+        'parseEnumAttr'                   // parseValueFromAttr
+      );
+    }
+
+    if (hvacFanControlEndpoint) {
+      const fanModeSeqProperty = this.addProperty(
+        node,                             // device
+        '_fanModeSeq',                    // name
+        {                                 // property description
+          label: 'Fan Sequence',
+          type: 'string',
+          readOnly: true,
+          enumArray: HVAC_FAN_SEQ,
```

Comment:
s/enumArray/enum/

We don't support `enumArray`.

Reply:
Right - I'm just using that at the adapter level. The enumArray never makes it to the gateway.

It would be useful to have a 'map' type where you can provide a dictionary of strings and values. The UI would show the strings but store the values.

Reply:
I think what you want is mozilla-iot/gateway#1310